x/cosmwasmpool/README.md

```diff
@@ -0,0 +1,284 @@
+# CosmWasm Pool
+
+The CosmWasm Pool Module is an extension for the Osmosis pools, aiming to create a custom module that allows users to create and manage liquidity pools backed by CosmWasm smart contracts. The feature enables developers to build and deploy custom smart contracts that can be integrated with the rest of the pool types on the Osmosis chain.
+
+The module is built on top of the CosmWasm smart contracting platform, which provides a secure and efficient way to develop and execute WebAssembly (Wasm) smart contracts on the Cosmos SDK.
+
+Having pools in CosmWasm provides several benefits, one of which is avoiding the need for chain upgrades when introducing new functionalities or modifying existing ones related to liquidity pools. This advantage is particularly important in the context of speed of development and iteration.
+
+An example of a CosmWasm pool type: [transmuter](https://github.com/osmosis-labs/transmuter)
+
+
+## Creating new CosmWasm Pool
+
+To create new CosmWasm Pool, there are 3 modules involved: `x/cosmwasmpool`, `x/wasm`, and `x/poolmanager`. Here is an overview of the process:
```

Comment:
The naming here seems odd. Shouldn't poolmanager be the entrypoint and then have to options cosmwasmpool or nativepool (or something like that)

Reply:
ahh, I get what you mean now! I guess what was confusing was the diagram. The one bellow clarified it for me.

Reply:
Or maybe making it explicit that this is an overview, sort of system context diagram. And for detail, see sequence diagram below...

Reply:
I think either of those works. My confusion was because I thought the Y part of the diagram meant you had to take one of the two paths